clang/lib/CIR/CodeGen/CIRGenFunction.cpp

```diff
@@ -234,6 +273,29 @@ cir::FuncOp CIRGenFunction::generateCode(clang::GlobalDecl gd, cir::FuncOp fn,
   return fn;
 }
 
+clang::QualType CIRGenFunction::buildFunctionArgList(clang::GlobalDecl gd,
+                                                     FunctionArgList &args) {
+  const auto *fd = cast<FunctionDecl>(gd.getDecl());
+  QualType retTy = fd->getReturnType();
+
+  const auto *md = dyn_cast<CXXMethodDecl>(fd);
+  if (md && md->isInstance())
+    cgm.errorNYI(fd->getSourceRange(), "buildFunctionArgList: CXXMethodDecl");
+
+  if (isa<CXXConstructorDecl>(fd))
```

Comment:
Do we handle `CXXConversionDecl` anywhere?  I'm trying to think whether it should have special handling.

Reply:
Not in this function. That is, not in either `CodeGenFunction::BuildFunctionArgList` or the incubator implementation of this function.